roles/sap_control/tasks/main.yml

```diff
@@ -1,133 +1,61 @@
-#     sap_control_sap_type: "{{ sap_control_function.split('_')[2] | lower }}"
-
-# - name: Set sap_facts_param
-#   ansible.builtin.set_fact:
-#     sap_facts_param: "{{ sap_control_get_type }}"
-
-# - name: Set sap_facts_param
-#   ansible.builtin.set_fact:
-#     sap_facts_param: "{{ sap_control_sap_type }}"
-#   when:
-#     - "'all' in sap_control_function"
-#     - "'sap' not in sap_control_sap_type"
+    - __sap_control_fact_instances is defined
+    - __sap_control_fact_instances | length > 0
 
-# # When not all
-# - name: Set sap_facts_param
-#   ansible.builtin.set_fact:
-#     sap_facts_param: "{{ sap_sid }}"
-#   when:
-#     - "'all' not in sap_control_function"
 
-# # Get SAP Info
-# - name: Get SAP Info
-#   vars:
-#     sap_info_get_function: "get_{{ sap_control_get_type }}_{{ sap_control_sap_type }}"
-#   ansible.builtin.include_role:
-#     name: roles/sap_info
-
-# # Get SAP Info
-# - name: Get SAP Info
-#   vars:
-#     sap_info_get_function: "get_{{ sap_control_get_type }}_{{ sap_control_sap_type }}"
-#   ansible.builtin.include_role:
-#     name: roles/sap_info
-
-# Get SAP Facts
-- name: Run sap_facts module to gather SAP facts
-  community.sap_libs.sap_system_facts:
-    # param: "{{ sap_facts_param }}"
-  register: sap_facts_register
-
-- name: Debug result from sap_libs.sap_system_facts
-  ansible.builtin.debug:
-    msg: "{{ sap_facts_register.ansible_facts.sap }}"
-    verbosity: 1
-
-- name: pause for 10 Seconds
-  ansible.builtin.wait_for:
-    timeout: 10
-
-# Debugging stuff
-- name: Display parameters for runtime
-  ansible.builtin.debug:
-    msg:
-      - "Starting sap_control with the following parameters: "
-      - "Function: {{ sap_control_function }}"
-      - "Standard commands:"
-      - "  Start: {{ sap_control_start }}"
-      - "  Stop: {{ sap_control_stop }}"
-      - "System commands (if applicable):"
-      - "  StartSystem: {{ sap_control_startsystem }}"
-      - "  StopSystem: {{ sap_control_stopsystem }}"
-      - "  RestartSystem: {{ sap_control_restartsystem }}"
-      - "  UpdateSystem: {{ sap_control_updatesystem }}"
-      - "  WaitforStopped: {{ sap_control_waitforstopped }}"
-      - "  WaitforStarted: {{ sap_control_waitforstarted }}"
-      - "NoWait: {{ nowait }}"
-
-# Start SAP Control
-- name: SAP Control
-  ansible.builtin.include_tasks: prepare.yml
-  loop: "{{ vars[sap_control_function + '_list'] }}"
-  loop_control:
-    loop_var: function_list
+- name: Block to execute only when commands are defined
+  when:
+    - __sap_control_fact_instances is defined
+    - __sap_control_fact_instances | length > 0
+    - __sap_control_commands is defined
+    - __sap_control_commands | length > 0
+  block:
+    - name: SAP Control - Show execution steps
+      ansible.builtin.debug:
+        msg: |
+          The Ansible Role `sap_control` is executed with function: '{{ sap_control_function }}'
+
+          Following commands will now be executed in this order by listed users:
```

Comment:
Following -> The following (same as in file roles/sap_control/tasks/detection/detect_sids.yml)